### PR TITLE
moved integration testcases to separate module

### PR DESCRIPTION
**Pull request by richarxt**
Moved all integration test cases from different modules to a new module.

### CODE REVIEW

File: blueflood-integration-tests/src/integration-test/java/com/rackspacecloud/blueflood/core/cache/MetadataCacheBatchedModeIntegrationTest.java [hunk 1]
```diff
@@ -14,21 +14,21 @@
  *    limitations under the License.
  */
 
-package com.rackspacecloud.blueflood.cache;
+package com.rackspacecloud.blueflood.core.cache;
```
Comment (zzantozz) [inline blueflood-integration-tests/src/integration-test/java/com/rackspacecloud/blueflood/core/cache/MetadataCacheBatchedModeIntegrationTest.java:17]: Why move to a different package? Tests should be in the same package as the source they're testing.

File: blueflood-integration-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+              <skipTests>${skip.integration.tests}</skipTests>
+              <includes>
+                <include>**/*Integration*.java</include>
+              </includes>
+              <excludes>
+                <!-- requires an API service -->
+                <exclude>**/InternalApiIntegrationTest.java</exclude>
+                <!-- this one requires a live zookeeper -->
+                <exclude>**/ZKShardLockManagerIntegrationTest.java</exclude>
+                <exclude>**/ScheduleContextIntegrationTest.java</exclude>
```
Comment (zzantozz) [inline blueflood-integration-tests/pom.xml:182]: We need to remember to take a look at these tests. If they're always turned off, what's the point of even having them?

**Review by zzantozz — COMMENTED** [review]
Some things to look at. Also consider that for a change like this, turning off "optimize imports" would help reduce the diff a lot and make much less to read through.

File: blueflood-integration-tests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    <testResources>
+      <testResource>
+        <directory>src/integration-test/java</directory>
+      </testResource>
+    </testResources>
+
+    <plugins>
+
+      <!-- test -->
+      <!-- Used to add source directories to our build. -->
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:62]: removing the coment

File: blueflood-http/pom.xml [hunk 4]
```diff
@@ -186,13 +97,12 @@
...
-      <scope>test</scope>
-      <optional>true</optional>
     </dependency>
 
     <dependency>
       <groupId>org.hamcrest</groupId>
       <artifactId>hamcrest-core</artifactId>
+      <version>1.3</version>
       <scope>test</scope>
     </dependency>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-http/pom.xml:107]: Check by removing this if it is auto pulled by hancrest-library

File: blueflood-core/pom.xml [hunk 5]
```diff
@@ -31,59 +31,13 @@
...
-      <testResource>
-        <directory>src/integration-test/java</directory>
-      </testResource>
-    </testResources>
-
     <plugins>
 
-      <!-- test -->
-
       <!-- Used to add source directories to our build. -->
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-core/pom.xml:40]: done

File: blueflood-core/pom.xml [hunk 6]
```diff
@@ -31,59 +31,13 @@
...
-                </resource>
-              </resources>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
-   
       <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-jar-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-core/pom.xml:43]: done

File: blueflood-core/pom.xml [hunk 7]
```diff
@@ -371,6 +277,8 @@
     <dependency>
       <groupId>org.hamcrest</groupId>
       <artifactId>hamcrest-library</artifactId>
+      <version>1.3</version>
+      <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-core/pom.xml:281]: reverted

File: blueflood-integration-tests/pom.xml [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+      <groupId>pl.pragmatists</groupId>
+      <artifactId>JUnitParams</artifactId>
+    </dependency>
+
+    <dependency>
+      <artifactId>blueflood-core</artifactId>
+      <groupId>com.rackspacecloud</groupId>
+      <version>${project.version}</version>
+      <type>test-jar</type>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:49]: right..removed

File: blueflood-integration-tests/pom.xml [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+  </dependencies>
+
+  <build>
+    <testResources>
+      <testResource>
+        <directory>src/integration-test/java</directory>
+      </testResource>
+    </testResources>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:57]: removed

File: blueflood-integration-tests/pom.xml [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+    <plugins>
+
+      <!-- test -->
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:61]: removed

File: pom.xml [hunk 11]
```diff
@@ -83,18 +73,6 @@
...
-        <artifactId>hamcrest-core</artifactId>
-        <version>1.3</version>
-        <scope>test</scope>
-      </dependency>
-      <dependency>
-        <groupId>org.hamcrest</groupId>
-        <artifactId>hamcrest-library</artifactId>
-        <version>1.3</version>
-        <scope>test</scope>
-      </dependency>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply pom.xml:97]: reverted

File: blueflood-integration-tests/src/integration-test/java/com/rackspacecloud/blueflood/core/cache/MetadataCacheBatchedModeIntegrationTest.java (hunk 1 above)
Comment (richarxt) [reply blueflood-integration-tests/src/integration-test/java/com/rackspacecloud/blueflood/core/cache/MetadataCacheBatchedModeIntegrationTest.java:17]: done

File: blueflood-integration-tests/.gitignore [hunk 12]
```diff
@@ -0,0 +1 @@
+tests.log.*
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/.gitignore:1]: done

File: blueflood-integration-tests/pom.xml [hunk 13]
```diff
@@ -0,0 +1,199 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:1]: done

File: blueflood-integration-tests/pom.xml [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>blueflood</artifactId>
+    <groupId>com.rackspacecloud</groupId>
+    <version>2.1.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>blueflood-integration-tests</artifactId>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:12]: Added

File: pom.xml [hunk 15]
```diff
@@ -302,7 +199,7 @@
           <cassandraDir>/tmp/bf-cassandra-${cassandra.plugin.version}-${maven.build.timestamp}/</cassandraDir>
           <loadFailureIgnore>false</loadFailureIgnore>
           <startNativeTransport>true</startNativeTransport>
-          <script>${project.basedir}/src/cassandra/cli/load.cdl</script>
+          <script>${main.basedir}/src/cassandra/cli/load.cdl</script>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply pom.xml:202]: done

File: blueflood-integration-tests/pom.xml (hunk 2 above)
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:182]: noted

File: blueflood-integration-tests/pom.xml [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+
+  <properties>
+    <main.basedir>${project.parent.basedir}</main.basedir>
```
[the remark this answers is not in the document]
Comment (richarxt) [reply blueflood-integration-tests/pom.xml:15]: done

**Pull request merged**